ci_framework/roles/repo_setup/tasks/main.yml

```diff
@@ -16,6 +16,8 @@
 
 - name: Install repo-setup
   ansible.builtin.import_tasks: install.yml
+- name: Include component repo if required
+  ansible.builtin.import_tasks: component_repo.yml
```

Comment:
Considering the task name ("if required"), the `when` condition from `ci_framework/roles/repo_setup/tasks/component_repo.yml` would be probably better moved here instead.